### PR TITLE
resgroup: use proper memory context when checking bypass mode

### PR DESCRIPTION
**Pull request by pivotal-ning-yu**
In shouldBypassQuery() we used to parse the query_string with the memory
context MessageContext, however when it is NULL we would crash at
runtime, this is possible by extensions or bgworkers when executing like
below:

    debug_query_string = "select 1";
    StartTransactionCommand();

This is not the recommended order to set debug_query_string, however we
should not put a constraint on the order by resource group anyway.

To fix this we create and switch to a temporary memory context when
MessageContext is NULL.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by pivotal-ning-yu** [comment]
We have passed resource group tests locally, the PR pipeline will run the ICW.

File: src/backend/utils/resgroup/resgroup.c [hunk 1]
```diff
@@ -3449,8 +3450,27 @@ shouldBypassQuery(const char *query_string)
...
 	/*
 	 * Switch to appropriate context for constructing parsetrees.
+	 *
+	 * It is possible that MessageContext is NULL, for example in a bgworker:
+	 *
+	 *     debug_query_string = "select 1";
+	 *     StartTransactionCommand();
+	 *
+	 * This is not the recommended order of setting debug_query_string, but we
+	 * should not put a constraint on the order by resource group anyway.
```
Comment (gkokolatos) [inline src/backend/utils/resgroup/resgroup.c]: Thanks for the detailed comment

**Comment by HaozhouWang** [comment]
checked on my local machine, no problem, +1

**Comment by pivotal-ning-yu** [comment]
Thanks for the reviews and verification, I'll merge it on master now, I'll also back-port it to 6X and 5X.

**Pull request merged**